### PR TITLE
MBST-13163: Added PA genre mapping to MetaBroadcast tags.

### PR DESCRIPTION
**Pull request by MartinsIrbe**
Added PA genre mapping to MetaBroadcast tags for brand, series and episode.

### CODE REVIEW

File: src/main/java/org/atlasapi/remotesite/pa/PaProgrammeProcessor.java [hunk 1]
```diff
@@ -498,6 +504,13 @@ private void setGenres(ProgData progData, Content content) {
         }
     }
 
+    private void setTags(ProgData progData, Content content) {
+        Set<String> tagsFromGenres = paTagMap.map(content.getGenres());
+        if (!tagsFromGenres.isEmpty()) {
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/pa/PaProgrammeProcessor.java]: Why do we need this guard?

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    private void mapNews() {
+        paTagMap.put("http://pressassociation.com/genres/2F02", "news-weather");
+        paTagMap.put("http://pressassociation.com/genres/2F03", "news-weather");
+        paTagMap.put("http://pressassociation.com/genres/2F04", "news-weather");
+        paTagMap.put("http://pressassociation.com/genres/2F05", "news-weather");
+        paTagMap.put("http://pressassociation.com/genres/2F06", "news-weather");
+    }
+
+    public Set<String> map(Set<String> genres) {
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: This needs documentation.

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
+package org.atlasapi.remotesite.pa;
+
+import java.util.Map;
+import java.util.Set;
+
+import com.google.api.client.util.Maps;
+import com.google.api.client.util.Sets;
+
+public class PaTagMap {
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: This needs documentation.

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+    public Set<String> map(Set<String> genres) {
+        Set<String> tags = Sets.newHashSet();
+        for (String genre : genres) {
+            if (genre.contains("http://pressassociation.com/genres/")) {
+                tags.add(paTagMap.get(genre));
+            }
+        }
+
+        // Checking if the tags set has only a one tag with the value - film, this means that the
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: this means that the... what?

File: src/test/java/org/atlasapi/remotesite/pa/PaTagMapTest.java [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+
+    private PaTagMap paTagMap;
+
+    @Before
+    public void setUp() {
+        paTagMap = new PaTagMap();
+    }
+
+    @Test
+    public void testTagMappingForFilms() {
```
Comment (emilssolmanis) [inline src/test/java/org/atlasapi/remotesite/pa/PaTagMapTest.java]: Most of these tests can be refactored and made simpler using [parameterized tests](https://github.com/junit-team/junit/wiki/Parameterized-tests).

File: src/main/java/org/atlasapi/remotesite/pa/PaProgrammeProcessor.java (hunk 1 above)
Comment (MartinsIrbe) [inline src/main/java/org/atlasapi/remotesite/pa/PaProgrammeProcessor.java]: To check if the mapped tags set isn't empty before adding tags to the content.

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java (hunk 2 above)
Comment (MartinsIrbe) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: :+1:

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java (hunk 3 above)
Comment (MartinsIrbe) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: :+1:

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java (hunk 4 above)
Comment (MartinsIrbe) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: the film is action by default, therefore we are adding `action` tag.

File: src/test/java/org/atlasapi/remotesite/pa/PaTagMapTest.java (hunk 5 above)
Comment (MartinsIrbe) [inline src/test/java/org/atlasapi/remotesite/pa/PaTagMapTest.java]: Good idea, however I haven't used parameterised tests before, will look into it when I got more time :+1:

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    private void mapNews() {
+        paTagMap.put("http://pressassociation.com/genres/2F02", "news-weather");
+        paTagMap.put("http://pressassociation.com/genres/2F03", "news-weather");
+        paTagMap.put("http://pressassociation.com/genres/2F04", "news-weather");
+        paTagMap.put("http://pressassociation.com/genres/2F05", "news-weather");
+        paTagMap.put("http://pressassociation.com/genres/2F06", "news-weather");
+    }
+
+    /**
+     * This methods maps PA content genres with MetaBroadcast tags.
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: The JavaDoc syntax mandates you to have the summary on the first line, this should complain if you try genning the docs. I.e., it should be 
```
/** Summary
  *
  * Optional extended description
  * 
  * @param ...
  */
```

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+import com.google.api.client.util.Maps;
+import com.google.api.client.util.Sets;
+
+/**
+ * This class is used for mapping PA content genres with MetaBroadcast tags.
+ */
+public class PaTagMap {
+
+    private final Map<String, String> paTagMap = Maps.newHashMap();
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: This should also be immutable, it isn't right now.

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java [hunk 8]
```diff
@@ -0,0 +1,176 @@
+package org.atlasapi.remotesite.pa;
+
+import java.util.Map;
+import java.util.Set;
+
+import com.google.api.client.util.Maps;
+import com.google.api.client.util.Sets;
+
+/**
+ * This class is used for mapping PA content genres with MetaBroadcast tags.
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: This is not proper documentation. Everything you've written here can be deduced from the class name. You should document the parts that can't, like *why* this exists and where / how it's used and link to those classes.

File: src/main/java/org/atlasapi/remotesite/pa/PaProgrammeProcessor.java (hunk 1 above)
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/remotesite/pa/PaProgrammeProcessor.java]: @MartinsIrbe Yeah, but presumably it's an empty set by default anyway, so if you just reset it to empty, that's a noop. Which would make the guard redundant. Or am I wrong?

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java (hunk 6 above)
Comment (MartinsIrbe) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: didn't know, will add it :)

File: src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java (hunk 8 above)
Comment (MartinsIrbe) [inline src/main/java/org/atlasapi/remotesite/pa/PaTagMap.java]: :+1: will include that

File: src/main/java/org/atlasapi/remotesite/pa/PaProgrammeProcessor.java (hunk 1 above)
Comment (MartinsIrbe) [inline src/main/java/org/atlasapi/remotesite/pa/PaProgrammeProcessor.java]: yes you are right, will remove it :+1:

**Comment by emilssolmanis** [comment]
lgtm. rebase before merging. :shipit:

**Pull request closed without merging**